tests/test_routing.py

```diff
@@ -669,6 +669,45 @@ def run_shutdown():
     assert shutdown_complete
 
 
+def test_lifespan_with_state(test_client_factory):
+    startup_complete = False
+    shutdown_complete = False
+
+    async def hello_world(request):
+        request.state.count += 1
+        request.state.list.append(1)
```

Comment:
```suggestion
        # modifications to the state should not leak across requests
        assert request.state.count == 0
        # modify the state, this should not leak to the lifespan or other requests
        request.state.count += 1
        # since state.list is a mutable object this modification _will_ leak across requests and to the lifespan
        request.state.list.append(1)
```